torch_object_detection_segmenter.py

```diff
@@ -1,31 +1,191 @@
+            :param img: :py:mod:`PIL.Image`, the image to be resized
+            :param target_size: desired output size. If size is a sequence like
+                (h, w), the output size will be matched to this. If size is an int,
+                the output will have the same height and width as the `target_size`.
+            :param top: the vertical coordinate of the top left corner of the crop box.
+            :param left: the horizontal coordinate of the top left corner of the crop box.
+            :param how: the way of cropping. Valid values include `center`, `random`, and, `precise`. Default is `precise`.
+                - `center`: crop the center part of the image
+                - `random`: crop a random part of the image
+                - `precise`: crop the part of the image specified by the crop box with the given ``top`` and ``left``.
+                .. warning:: When `precise` is used, ``top`` and ``left`` must be fed valid value.
+            """
+            import PIL.Image as Image
+            assert isinstance(img, Image.Image), 'img must be a PIL.Image'
+            img_w, img_h = img.size
+            if isinstance(target_size, int):
+                target_h = target_w = target_size
+            elif isinstance(target_size, Tuple) and len(target_size) == 2:
+                target_h, target_w = target_size
+            else:
+                raise ValueError(f'target_size should be an integer or a tuple of two integers: {target_size}')
+            w_beg = left
+            h_beg = top
+            if how == 'center':
+                w_beg = int((img_w - target_w) / 2)
+                h_beg = int((img_h - target_h) / 2)
+            elif how == 'random':
+                w_beg = np.random.randint(0, img_w - target_w + 1)
+                h_beg = np.random.randint(0, img_h - target_h + 1)
+            elif how == 'precise':
+                assert (w_beg is not None and h_beg is not None)
+                assert (0 <= w_beg <= (img_w - target_w)), f'left must be within [0, {img_w - target_w}]: {w_beg}'
+                assert (0 <= h_beg <= (img_h - target_h)), f'top must be within [0, {img_h - target_h}]: {h_beg}'
+            else:
+                raise ValueError(f'unknown input how: {how}')
+            if not isinstance(w_beg, int):
+                raise ValueError(f'left must be int number between 0 and {img_w}: {left}')
+            if not isinstance(h_beg, int):
+                raise ValueError(f'top must be int number between 0 and {img_h}: {top}')
+            w_end = w_beg + target_w
+            h_end = h_beg + target_h
+            img = img.crop((w_beg, h_beg, w_end, h_end))
+            return img, h_beg, w_beg
+
         if not docs:
             return
-        for doc in docs:
-            assert doc.blob.shape[0] == 256
-            doc.embedding = np.random.rand(self._dim)
\ No newline at end of file
+
+        batch = docs.get_attributes('blob') # (2, 681, 1264, 3) with imgs/cars.jpg
```

Comment:
this is confusing, you are not using batching in this case. Is it not possible for the model to predict in batches?

Reply:
It is possible, but then u are not using batching and therefore this model will possibly explode

Reply:
isn't the model already predicting in batches taking in a batch of image blobs and outputing batched predictions ?

Reply:
review what is batching and meaning of batching size. Here it is predicting for the complete set of documents at once

Reply:
But I added the batch generator that yield data[i: i + batch_size], a subset of the documents